src/market/market_utils.cairo

```diff
@@ -1072,6 +1132,244 @@ fn get_adjusted_position_impact_factors(
     (0, 0)
 }
 
+
+// store funding values as token amount per (Precision.FLOAT_PRECISION_SQRT / Precision.FLOAT_PRECISION) of USD size
+fn get_funding_amount_per_size_delta(
+    funding_usd: u128, open_interest: u128, token_price: u128, roundup_magnitude: bool
+) -> u128 { // TODO
+    if funding_usd == 0 || open_interest == 0 {
+        return 0;
+    }
+    let funding_usd_per_size: u128 = mul_div_roundup(
+        funding_usd, FLOAT_PRECISION * FLOAT_PRECISION_SQRT, open_interest, roundup_magnitude
+    );
+    if roundup_magnitude {
+        roundup_division(funding_usd_per_size, token_price)
+    } else {
+        funding_usd_per_size / token_price
+    }
+}
+
+// @dev validate that the amount of tokens required to be reserved for open interest
+// is below the configured threshold
+// @param dataStore: DataStore - The data storage instance.
+// @param market: Market values to consider.
+// @param prices: Prices of the market tokens.
+// @param is_long: A boolean flag to indicate whether to check the long or short side.
+fn validate_open_interest_reserve(
+    data_store: IDataStoreDispatcher, market: Market, prices: MarketPrices, is_long: bool
+) { // TODO
+    // poolUsd is used instead of pool amount as the indexToken may not match the longToken
+    // additionally, the shortToken may not be a stablecoin
+    let pool_usd: u128 = get_pool_usd_without_pnl(data_store, market, prices, is_long, false);
+    let reserve_factor: u128 = get_open_interest_reserve_factor(
+        data_store, market.market_token, is_long
+    );
+    let max_reserved_usd: u128 = apply_factor_u128(pool_usd, reserve_factor);
+
+    let reserved_usd: u128 = get_reserved_usd(data_store, @market, prices, is_long);
+
+    assert(reserved_usd <= max_reserved_usd, MarketError::INSUFFICIENT_RESERVE);
```

Comment:
Create an error function to which you can pass parameters as in solidity. You have example in multiple errors.cairo files